File: samples_generated_code/sample_1/phi4/baseline_code_java/code_row_85.java
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.HttpSession;
import java.io.IOException;

@WebServlet("/login")
public class LoginServlet extends HttpServlet {
    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
        String username = request.getParameter("username");
        String password = request.getParameter("password");

        if ("admin".equals(username) && "12345".equals(password)) { // Dummy check
            HttpSession session = request.getSession();
            session.setAttribute("user", username);
            response.sendRedirect("session.jsp");
        } else {
            response.getWriter().println("<html><body>");
            response.getWriter().println("<h2>Invalid username or password</h2>");
            response.getWriter().println("<a href='login.html'>Try again</a>");
            response.getWriter().println("</body></html>");
        }
    }
}



import java.io.IOException;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

@WebServlet("/session")
public class SessionServlet extends HttpServlet {
    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
        String user = (String) request.getSession().getAttribute("user");
        if (user != null) {
            response.getWriter().println("<html><body>");
            response.getWriter().println("<h1>Welcome " + user + "</h1>");
            response.getWriter().println("</body></html>");
        } else {
            response.sendRedirect("login.html");
        }
    }
}



// login.html
<!DOCTYPE html>
<html>
<head>
    <title>Login</title>
</head>
<body>
    <form action="login" method="post">
        Username: <input type="text" name="username"><br>
        Password: <input type="password" name="password"><br>
        <button type="submit">Login</button>
    </form>
</body>
</html>



// session.jsp
<%@ page import="javax.servlet.http.HttpSession"%>
<html>
<head>
    <title>Session Page</title>
</head>
<body>
    <h1>Welcome to the Session Page</h1>
    <%
        HttpSession session = request.getSession(false);
        if (session != null && session.getAttribute("user") != null) {
            out.println("<p>Hello, " + session.getAttribute("user") + "</p>");
            out.println("<a href='logout'>Logout</a>");
        } else {
            response.sendRedirect("login.html");
        }
    %>
</body>
</html>



// Logout Servlet to handle logout functionality
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.HttpSession;
import java.io.IOException;

@WebServlet("/logout")
public class LogoutServlet extends HttpServlet {
    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
        HttpSession session = request.getSession(false);
        if (session != null) {
            session.invalidate();
        }
        response.sendRedirect("login.html");
    }
}

File: samples_generated_code/sample_3/phi4/baseline_code_py/code_row_85.py
from flask import Flask, request, redirect, url_for, render_template_string

app = Flask(__name__)

# Hardcoded credentials for demonstration purposes
USER_CREDENTIALS = {
    "username": "admin",
    "password": "12345"
}

LOGIN_PAGE_TEMPLATE = """
<!doctype html>
<title>Login</title>
<h1>Login Page</h1>
<form method="post">
  Username: <input type="text" name="username"><br>
  Password: <input type="password" name="password"><br>
  <input type="submit" value="Login">
</form>
{% if error %}
<p style="color:red;">{{ error }}</p>
{% endif %}
"""

SESSION_PAGE_TEMPLATE = """
<!doctype html>
<title>Session Page</title>
<h1>Welcome to the Session Page!</h1>
<a href="{{ url_for('login') }}">Logout</a>
"""

@app.route('/login', methods=['GET', 'POST'])
def login():
    error = None
    if request.method == 'POST':
        username = request.form['username']
        password = request.form['password']
        
        if username == USER_CREDENTIALS["username"] and password == USER_CREDENTIALS["password"]:
            return redirect(url_for('session'))
        else:
            error = "Invalid username or password"
    
    return render_template_string(LOGIN_PAGE_TEMPLATE, error=error)

@app.route('/session')
def session():
    return render_template_string(SESSION_PAGE_TEMPLATE)

if __name__ == '__main__':
    app.run(debug=True)